### PR TITLE
[Opt] make a loop vectorized in hll serialize

### PR DESCRIPTION
**Pull request by zbtzbtzbt**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Problem Summary(Required) ：

make a heavy loop vectorized

### CODE REVIEW

**Review by imay — DISMISSED** [review]
LGTM

File: be/src/storage/hll.cpp [hunk 1]
```diff
@@ -202,9 +202,7 @@ size_t HyperLogLog::serialize(uint8_t* dst) const {
     case HLL_DATA_FULL: {
         uint32_t num_non_zero_registers = 0;
         for (int i = 0; i < HLL_REGISTERS_COUNT; i++) {
-            if (_registers.data[i] != 0) {
-                num_non_zero_registers++;
-            }
+            num_non_zero_registers += (_registers.data[i] != 0);
```
Comment (mofeiatwork) [inline be/src/storage/hll.cpp:205]: What's the difference of assembly?

**Review by Seaven — DISMISSED** [review]

File: be/src/storage/hll.h [hunk 2]
```diff
@@ -238,8 +238,6 @@ class HllSetResolver {
 public:
     HllSetResolver()
 
-    {}
-
```
Comment (Seaven) [inline be/src/storage/hll.h:242]: You can't delete `{}`, compile is failed

Comment (zbtzbtzbt) [reply be/src/storage/hll.h]: done

File: be/src/storage/hll.cpp (hunk 1 above)
Comment (zbtzbtzbt) [reply be/src/storage/hll.cpp:205]: after vectoriezd, there some xmm assembly

**Review by Seaven — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**